### PR TITLE
Refactor Labels CRD

### PR DESCRIPTION
**Pull request by slintes**
Removed rules, use string map for labels.
Also remove "kni" from projects name (leftover from migration to new org), and some smaller comment/doc fixes

### CODE REVIEW

File: pkg/nodes.go [hunk 1]
```diff
@@ -50,35 +49,27 @@ func AddLabels(node *v1.Node, labels v1beta1.Labels, log logr.Logger) bool {
...
+		if err != nil {
+			log.Error(err, "Invalid regular expression, moving on to next rule")
+			continue
+		}
+		if !match {
+			continue
+		}
+		// we have a match, add labels!
+		for name, value := range labels.Spec.Labels {
+			if val, ok := node.Labels[name]; !ok || val != value {
```
Comment (n1r1) [inline pkg/nodes.go:70]: I think that `node.Labels` could be `nil` (although very unlikely) and `node.Labels[name]` might fail.

File: pkg/nodes.go [hunk 2]
```diff
@@ -50,35 +49,27 @@ func AddLabels(node *v1.Node, labels v1beta1.Labels, log logr.Logger) bool {
...
+				log.Info("Adding label to node based on pattern", "labelName", name, "labelValue", value, "pattern", nodeNamePattern)
```
Comment (n1r1) [inline pkg/nodes.go]: maybe add node name?

**Review by n1r1 — COMMENTED** [review]
I like the way it looks now. It simplifies the code.

File: pkg/nodes.go (hunk 1 above)
Comment (slintes) [reply pkg/nodes.go:70]: Good catch, I moved up the check which followed after this line

File: pkg/nodes.go (hunk 2 above)
Comment (slintes) [reply pkg/nodes.go]: done

File: PROJECT [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
 layout: go.kubebuilder.io/v3
-projectName: node-label-operator-kni
+projectName: node-label-operator
 repo: github.com/openshift-kni/node-label-operator
 resources:
 - crdVersion: v1
   group: node-labels
   kind: Labels
   version: v1beta1
-  webhookVersion: v1
```
Comment (beekhof) [inline PROJECT:10]: No webhook anymore?

File: api/v1beta1/labels_types.go [hunk 4]
```diff
@@ -24,18 +24,13 @@ import (
...
-}
-
-type Rule struct {
-	// NodeNames defines a list of node name patterns for which the given labels should be set.
+	// NodeNamePatterns defines a list of node name regex patterns for which the given labels should be set.
 	// String start and end anchors (^/$) will be added automatically
 	NodeNamePatterns []string `json:"nodeNamePatterns"`
 
 	// Label defines the labels which should be set if one of the node name patterns matches
 	// Format of label must be domain/name=value
```
Comment (beekhof) [inline api/v1beta1/labels_types.go:32]: Need to update description to match :)

**Pull request merged**